tmd/Topology/persistent_properties.py

```diff
@@ -0,0 +1,178 @@
+"""Persistent properties classes
+"""
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+class PersistentProperty(ABC):
+    """Abstract class for persistent properties that are defined
+    on persistent components.
+    """
+    @abstractmethod
+    def get(self, component_start):
+        """Get component property"""
+
+    @abstractmethod
+    def infinite_component(self, component_start):
+        """Get property for infinite component"""
+
+
+class NoProperty(PersistentProperty):
+    """Function class for extracting a barcode without properties
+    """
+    def __init__(self, _):
+        pass
+
+    def get(self, _):
+        """Returns empty list, does not contribute to component"""
+        return []
+
+    def infinite_component(self, _):
+        """Returns empty list, does not contribute to component"""
+        return []
+
+
+class PersistentAngles(PersistentProperty):
+    """Bifurcation angles per component
+    Args:
+        tree (Tree): A tree object
+    """
+    def __init__(self, tree):
+
+        section_begs, _ = tree.sections
+        section_parents, section_children = tree.parents_children
+
+        self._angles = get_angles(
+            tree, section_begs, section_parents, section_children)
+
+    def get(self, component_start):
+        """
+        Args:
+            component_start (int): The component start
+        Returns:
+            component_angles (list): A list of 4 angles
+        """
+        return self._angles[component_start]
+
+    def infinite_component(self, _):
+        """Given that there are not angles for the inf
+        component, nans are returned.
+        """
+        return [np.nan, np.nan, np.nan, np.nan]
+
+
+def _section_mean_radii(tree_diameters, section_begs, section_ends):
```

Comment:
Well, in order to be a method of persistent properties, the latter must store the tree, which is not the case here.

Reply:
Hmm why? The tree can remain in the parameters of the method, so it is not stored in the class instance.
But anyway, I think it could at least be a `@staticmethod` of `PersistentMeanRadius` to make it clear that it is only used in this context. So in the constructor you would call it with `self._section_mean_radii()` but `self` would not be passed as argument.

Reply:
kk, makes sense indeed